Python/Product/PythonTools/PythonTools/SearchPathManager.cs

```diff
@@ -41,10 +41,6 @@ class SearchPathManager : IVsFileChangeEvents, IDisposable {
             GC.SuppressFinalize(this);
```

Comment:
You can make this class `sealed`, remove `GC.SuppressFinalize(this);` and remove `protected virtual void Dispose` overload.
This is pretty much everything you need:
```cs
public void Dispose() {
    lock (_paths) {
        foreach (var p in _paths) {
            Unwatch(p.Cookie);
        }
        _paths.Clear();
    }

    var timer = _notifyChangeTimer;
    if (timer != null) {
        timer.Dispose();
    }
}
```

Reply:
Last time you suggested changes to IDisposable pattern, @zooba was saying that our static code analysis rules were complaining about this.

How do I run static code analysis on my local machine? Enabling it on the project causes tons of errors "The rule X in PTVS\Build\xTVS.ruleset could not be found"

Reply:
I'll change it, and if we get warnings we can change it back.